sound/soc/sof/ipc4-loader.c

```diff
@@ -265,8 +265,86 @@ static int sof_ipc4_query_fw_configuration(struct snd_sof_dev *sdev)
 	return sof_ipc4_fw_parse_ext_man(sdev, plat_data->fw, 0, 0);
 }
 
+static int sof_ipc4_load_library(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
+{
+	struct sof_ipc4_fw_data *ipc4_data = sdev->private;
+	struct snd_sof_pdata *plat_data = sdev->pdata;
+	const guid_t *widget_uuid = &swidget->uuid;
+	const char *fw_filename;
+	size_t fw_offset;
+	int ret, i;
+
+	/* nothing to do if widget UUID is not set */
+	if (!guid_is_null(&swidget->uuid))
+		return 0;
```

Comment:
again is this not an error? When could this happen?

Reply:
no this is not an error. Pipeline widgets do no have UUIDs